sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,464 @@
+
+		if (spender) {
+			const spenderContract = this.sdk.context.contracts[spender];
+			if (spenderContract) spenderAddress = spenderContract.address;
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'approve', [
+			spenderAddress,
+			amount,
+		]);
+	}
+
+	public redeemToken(token: ContractName) {
+		const tokenContract = this.sdk.context.contracts[token];
+
+		if (!tokenContract) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(tokenContract, 'redeem', [
+			this.sdk.context.walletAddress,
+		]);
+	}
+
+	public redeemVKwenta() {
+		return this.redeemToken('vKwentaRedeemer');
+	}
+
+	public redeemVeKwenta() {
+		return this.redeemToken('veKwentaRedeemer');
+	}
+
+	public vestToken(ids: number[]) {
+		const { RewardEscrow } = this.sdk.context.contracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		return this.sdk.transactions.createContractTxn(RewardEscrow, 'vest', [ids]);
+	}
+
+	public stakeKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('stake', amount);
+	}
+
+	public async unstakeKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('unstake', amount);
+	}
+
+	public async stakeEscrowedKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('stake', amount, { escrow: true });
+	}
+
+	public async unstakeEscrowedKwenta(amount: string | ethers.BigNumber) {
+		return this.performStakeAction('unstake', amount, { escrow: true });
+	}
+
+	public async getClaimableRewards(periods: number[]) {
+		const { MultipleMerkleDistributor } = this.sdk.context.contracts;
+
+		if (!MultipleMerkleDistributor) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const MultipleMerkleDistributorContract = new EthCallContract(
+			MultipleMerkleDistributor.address,
+			MultipleMerkleDistributorABI
+		);
+
+		const fileNames = periods
+			.slice(0, -1)
+			.map(
+				(i) =>
+					`trading-rewards-snapshots/${
+						this.sdk.context.networkId === 420 ? `goerli-` : ''
+					}epoch-${i}.json`
+			);
+
+		const responses: EpochData[] = [];
+
+		for (const fileName of fileNames) {
+			const response = await client.get(fileName);
+			responses.push(response.data ?? null);
+		}
+
+		const rewards = responses
+			.map((d, period) => {
+				const { walletAddress } = this.sdk.context;
+				const walletReward = d.claims[walletAddress];
+				return [walletReward.index, walletAddress, walletReward.amount, walletReward.proof, period];
+			})
+			.filter((x): x is ClaimParams => !!x);
+
+		const claimed: boolean[] = await this.sdk.context.multicallProvider.all(
+			rewards.map((reward) => MultipleMerkleDistributorContract.isClaimed(reward[0], reward[4]))
+		);
+
+		const { totalRewards, claimableRewards } = rewards.reduce(
+			(acc, next, i) => {
+				if (!claimed[i]) {
+					acc.claimableRewards.push(next);
+					acc.totalRewards += Number(next[2]) / 1e18;
```

Comment:
Should probably use Wei here so we remain consistent with how we treat big numbers and just to keep on the safe side of floating point issues.

Reply:
Yeah. Will replace with wei.